core/wasm/src/prepare.rs

```diff
@@ -29,38 +24,31 @@ impl<'a> ContractModule<'a> {
         })
     }
 
-    fn externalize_mem(&mut self) {
+    fn standardize_mem(&mut self) {
         let mut module = self
             .module
             .take()
             .expect("On entry to the function `module` can't be `None`; qed");
 
         let mut tmp = MemorySection::default();
 
-        let entry = module.memory_section_mut()
+        module.memory_section_mut()
             .unwrap_or_else(|| &mut tmp)
             .entries_mut()
             .pop();
 
-        match entry {
-            Some(mut entry) => {
-                if entry.limits().maximum().is_none() {
-                    entry = elements::MemoryType::new(
-                        entry.limits().initial(),
-                        Some(self.config.max_memory_pages));
-                }
+        let entry = elements::MemoryType::new(
+            self.config.initial_memory_pages,
+            Some(self.config.max_memory_pages));
 
-                let mut builder = builder::from_module(module);
-                builder.push_import(elements::ImportEntry::new(
-                    "env".to_owned(),
-                    "memory".to_owned(),
-                    elements::External::Memory(entry),
-                ));
+        let mut builder = builder::from_module(module);
+        builder.push_import(elements::ImportEntry::new(
+            "env".to_owned(),
```

Comment:
I think it's more idiomatic to use `to_string()`

Reply:
done